xonsh/completers/completer.py

```diff
@@ -40,3 +50,65 @@ def complete_completer(prefix, line, start, end, ctx):
         else:
             raise StopIteration
     return {i for i in possible if i.startswith(prefix)}
+
+
+def add_one_completer(name, func, loc="end"):
+    new = collections.OrderedDict()
```

Comment:
We don't need to necessarily tackle this just now, but I wonder if we even need `OrderedDict` here anymore now that dictionary order is guaranteed?

Reply:
the orderness was advertised in py3.6 I think. But I am not sure we can depend on it and it won't change in future. The other point, could you explain ?

Reply:
It was advertised in Python 3.6 as an implementation detail, but it was added to the language spec in 3.7, it isn't going anywhere

Reply:
still it is better to have this. Also I haven't added this one, I just renamed the function name. 

https://stackoverflow.com/questions/50872498/will-ordereddict-become-redundant-in-python-3-7#:~:text=No%20it%20won't%20become,an%20order%20dependent%20method%2C%20OrderedDict.